src/Menu/Provider/GroupMenuProvider.php

```diff
@@ -13,190 +13,69 @@
-                $menuItem->setAttribute('class', 'keep-open');
-                $menuItem->setExtra('keep_open', $group['keep_open']);
-            }
-        } elseif (1 === \count($group['items'])) {
-            if ($this->canGenerateMenuItem($group['items'][0], $group)) {
-                $menuItem = $this->generateMenuItem($group['items'][0], $group);
-                $menuItem->setExtra('on_top', $group['on_top']);
-            } else {
-                $menuItem->setDisplay(false);
-            }
+        public function get(string $name, array $options = []): ItemInterface
+        {
+            return $this->doGet($name, $options);
         }
-        $menuItem->setLabel($group['label']);
 
-        return $menuItem;
+        public function has(string $name, array $options = []): bool
+        {
+            return $this->doHas($name, $options);
+        }
     }
-
+} else {
     /**
-     * Checks whether a menu exists in this provider.
+     * Menu provider based on group options.
      *
-     * @param string $name
+     * @final since sonata-project/admin-bundle 3.52
      *
-     * @return bool
+     * @author Alexandru Furculita <alex@furculita.net>
      */
-    public function has($name, array $options = [])
-    {
-        return 'sonata_group_menu' === $name;
-    }
-
-    private function canGenerateMenuItem(array $item, array $group): bool
+    class GroupMenuProvider extends BaseGroupMenuProvider
     {
-        if (isset($item['admin']) && !empty($item['admin'])) {
-            $admin = $this->pool->getInstance($item['admin']);
-
-            // skip menu item if no `list` url is available or user doesn't have the LIST access rights
-            return $admin->hasRoute('list') && $admin->hasAccess('list');
-        }
-
-        //NEXT_MAJOR: Remove if statement of null checker.
-        if (null === $this->checker) {
-            return true;
-        }
-
-        // Making the checker behave affirmatively even if it's globally unanimous
-        // Still must be granted unanimously to group and item
-
-        $isItemGranted = true;
-        if (!empty($item['roles'])) {
-            $isItemGranted = false;
-            foreach ($item['roles'] as $role) {
-                if ($this->checker->isGranted($role)) {
-                    $isItemGranted = true;
-                    break;
-                }
-            }
-        }
-
-        $isGroupGranted = true;
-        if (!empty($group['roles'])) {
-            $isGroupGranted = false;
-            foreach ($group['roles'] as $role) {
-                if ($this->checker->isGranted($role)) {
-                    $isGroupGranted = true;
-                    break;
-                }
-            }
+        /**
+         * Retrieves the menu based on the group options.
+         *
+         * @param string $name
+         *
+         * @throws \InvalidArgumentException if the menu does not exist
+         *
+         * @return \Knp\Menu\ItemInterface
```

Comment:
This interface is already imported.
```suggestion
         * @return ItemInterface
```